test/envoye2e/driver/envoy.go

```diff
@@ -145,3 +156,60 @@ func getAdminPort(bootstrap string) (uint32, error) {
 	}
 	return port.PortValue, nil
 }
+
+// downloads env based on the given branch name. Return address of downloaded envoy.
+func downloadEnvoy(ver string) (string, error) {
+	proxyDepURL := fmt.Sprintf("https://raw.githubusercontent.com/istio/istio/release-%v/istio.deps", ver)
+	resp, err := http.Get(proxyDepURL)
+	if err != nil {
+		return "", fmt.Errorf("cannot get envoy sha from %v: %v", proxyDepURL, err)
+	}
+	defer resp.Body.Close()
+	istioDeps, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("cannot read body of istio deps: %v", err)
+	}
+
+	var deps []interface{}
+	if err := json.Unmarshal(istioDeps, &deps); err != nil {
+		return "", err
+	}
+	proxySHA := ""
+	for _, d := range deps {
+		if dm, ok := d.(map[string]interface{}); ok && dm["repoName"].(string) == "proxy" {
+			proxySHA = dm["lastStableSHA"].(string)
+		}
+	}
+
+	// make temp directory to put downloaded envoy binary.
```

Comment:
Return error if proxy_sha is still empty.

Reply:
Done.